WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergVideoPressUploadProcessor.swift

```diff
@@ -0,0 +1,174 @@
+import Foundation
+import Aztec
+
+class GutenbergVideoPressUploadProcessor: Processor {
+
+    let mediaUploadID: Int32
+    let serverMediaID: Int
+    let videoPressGUID: String
+    var videoPressURL: String = ""
+
+    private struct VideoPressBlockKeys {
+        static var name = "wp:videopress/video"
+        static var id = "id"
+        static var guid = "guid"
+        static var resizeToParent = "resizeToParent"
+        static var cover = "cover"
+        static var autoplay = "autoplay"
+        static var controls = "controls"
+        static var loop = "loop"
+        static var muted = "muted"
+        static var playsinline = "playsinline"
+        static var poster = "poster"
+        static var preload = "preload"
+        static var seekbarColor = "seekbarColor"
+        static var seekbarPlayedColor = "seekbarPlayedColor"
+        static var seekbarLoadingColor = "seekbarLoadingColor"
+        static var useAverageColor = "useAverageColor"
+    }
+
+    private struct VideoPressURLQueryParams {
+        static var resizeToParent = "resizeToParent"
+        static var cover = "cover"
+        static var autoPlay = "autoPlay"
+        static var controls = "controls"
+        static var loop = "loop"
+        static var muted = "muted"
+        static var persistVolume = "persistVolume"
+        static var playsinline = "playsinline"
+        static var posterUrl = "posterUrl"
+        static var preloadContent = "preloadContent"
+        static var sbc = "sbc"
+        static var sbpc = "sbpc"
+        static var sblc = "sblc"
+        static var useAverageColor = "useAverageColor"
+    }
```

Comment:
It would reduce the boilerplate a bunch if we used enums instead that are of type `String`. It would simply infer the ones that are the same with their constant names. And we can only specify the different ones.

```Swift
enum VideoPressBlockKeys: String {
    case name = "wp:videopress/video"
    case id = "id"
    case guid = "guid"
    case resizeToParent = "resizeToParent"
}
```

And the usage would be:
```Swift
print(VideoPressBlockKeys.name.rawValue)
print(VideoPressBlockKeys.id.rawValue)
```

Reply:
Thank you so much @alpavanoglu for taking a look at the PR and the feedback 🙇 !

> It would reduce the boilerplate a bunch if we used enums instead that are of type String.

Yep, using enums was my first option when I started working on this. However, I noticed that the rest of the upload processors ([example](https://github.com/wordpress-mobile/WordPress-iOS/blob/7cc5d89cc30d16a3dc50d88c8a797749f805ed38/WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergGalleryUploadProcessor.swift#L22-L29)) were using this approach so I followed the same path. Nevertheless, I agree that enums will reduce definitely the boilerplate.

> It would simply infer the ones that are the same with their constant names. And we can only specify the different ones.

Great! I understand following the example you shared that we won't need to define the String for some of them, i.e.:
```
enum VideoPressBlockKeys: String {
    case name = "wp:videopress/video"
    case id
    case guid
    case resizeToParent
}
```

Is this accurate?